src/coreclr/vm/jitinterface.cpp

```diff
@@ -11915,12 +11956,25 @@ bool CEEInfo::getReadonlyStaticFieldValue(CORINFO_FIELD_HANDLE fieldHnd, uint8_t
             if (fieldObj != NULL)
             {
                 Object* obj = OBJECTREFToObject(fieldObj);
+                size_t handle;
                 if (GCHeapUtilities::GetGCHeap()->IsInFrozenSegment(obj))
                 {
-                    intptr_t ptr = (intptr_t)obj;
-                    memcpy(buffer, &ptr, sizeof(intptr_t));
-                    result = true;
+                    handle = (size_t)obj;
+                }
+                else
+                {
+                    PTR_MethodTable objMT = obj->GetMethodTable();
+                    // We don't need to limit it to these types but JIT doesn't need other types of
+                    // objects at the moment so it's for better throughput.
```

Comment:
We may want to add bool argument to `getReadonlyStaticFieldValue` that says whether the JIT is interested in non-frozen object handles.